### PR TITLE
Ruby 2.6 (LG-2547)

### PR DESCRIPTION
**Pull request by zachmargolis**
Upgrading!

Thanks for @jmhooper for pointing me in the direction of the OpenSSL issue!

We will likely need a devops change as well, Ruby 2.7 is beyond what can currently be deployed, but @jgrevich said they could help with that

### CODE REVIEW

**Comment by jmhooper** [comment]
Looks like we also need to update the rubocop engine to 2.7

File: .rubocop.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ AllCops:
     - 'lib/deploy/*'
     - 'node_modules/**/*'
     - 'vendor/**/*'
-  TargetRubyVersion: 2.3
+  TargetRubyVersion: 2.7
```
Comment (jmhooper) [inline .rubocop.yml]: Huh, looks like you updated the target ruby version here. Not sure why CodeClimate is complaining...

Comment (zachmargolis) [reply .rubocop.yml]: Turns out our version of Rubocop was too old, so I upgraded it

**Comment by zachmargolis** [comment]
Do we have any strong opinions on language-level deprecation warnings? There are a *ton* of these printed when running the test suite, and I suspect we'll have to upgrade a lot of gems (including Rails) to get them all to go away.

```
~/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/json-2.2.0/lib/json/common.rb:156: warning: Using the last argument as keyword parameters is deprecated
~/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bullet-6.0.2/lib/bullet.rb:41: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
~/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/core_ext/module/delegation.rb:154: warning: The called method `delegate' is defined here
```

It's making me consider moving to something like 2.6 instead? I think we have devops support for 2.6 as-is as well.

**Comment by jmhooper** [comment]
Fun, it should be easy to go back down to 2.6, right? Just need to change the number and the rest of these changes still work

**Comment by zachmargolis** [comment]
@jmhooper Looks like things are passing, except the codeclimate diff-coverage, is that ok to skip here? Since there aren't any code changes?

**Comment by bleachbyte** [comment]
We (@jgrevich and I) poked this for a few hours; this PR seems to do the best work at correcting our current issue -- 18F/identity-devops#1951 -- which is blocking `identity-pivcac` hosts from coming up correctly with our current build/AMI. There are, however, `circleci` failures to be corrected:

```
Finished in 12.79 seconds (files took 7.5 seconds to load)
128 examples, 2 failures

Failed examples:

rspec ./spec/controllers/identify_controller_spec.rb:283 # IdentifyController GET / with a good referrer with a good certificate with a certificate ocsp error returns a token as ocsp error
rspec ./spec/controllers/identify_controller_spec.rb:260 # IdentifyController GET / with a good referrer with a good certificate with a certificate timeout returns a token as timeout
```

Our branches (since closed) generated _**20**_ errors, instead of 2, so we're deferring to this PR for verifying that `identity-pki` is working properly, at which point we can verify that `identity-pivcac` hosts are coming up correctly from our code in `identity-devops`.

**Comment by zachmargolis** [comment]
@bleachbyte I'm still trying to get more info, but those controller are flaky (confirmed by @jmhooper) so I'm just kicking off a rebuild

**Pull request merged**